ecl/resource_ecl_network_load_balancer_v2.go

```diff
@@ -0,0 +1,1384 @@
+		Timeout:      d.Timeout(schema.TimeoutUpdate),
+		Delay:        5 * time.Second,
+		PollInterval: loadBalancerUpdatePollInterval,
+		MinTimeout:   3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"error waiting for Load Balancer Syslog Server (%s) to become ACTIVE(after Syslog Server update): %s",
+			id, err)
+	}
+
+	return nil
+}
+
+func getLoadBalancerUpdateOpts(d *schema.ResourceData, gatewayInitialized bool, planUpdated bool) *load_balancers.UpdateOpts {
+	var returnUpdateOpts bool
+
+	var updateOpts load_balancers.UpdateOpts
+
+	s := d.Get("default_gateway").(string)
+	if d.HasChange("default_gateway") || (gatewayInitialized && s != "") {
+		returnUpdateOpts = true
+		var defaultGateway interface{}
+		if s == "" {
+			defaultGateway = nil
+		} else {
+			defaultGateway = s
+		}
+		updateOpts.DefaultGateway = &defaultGateway
+	}
+
+	if d.HasChange("description") {
+		returnUpdateOpts = true
+		description := d.Get("description").(string)
+		updateOpts.Description = &description
+	}
+
+	if d.HasChange("load_balancer_plan_id") && !planUpdated {
+		returnUpdateOpts = true
+		loadBalancerPlanID := d.Get("load_balancer_plan_id").(string)
+		updateOpts.LoadBalancerPlanID = loadBalancerPlanID
+	}
+
+	if d.HasChange("name") {
+		returnUpdateOpts = true
+		name := d.Get("name").(string)
+		updateOpts.Name = &name
+	}
+
+	if returnUpdateOpts {
+		return &updateOpts
+	} else {
+		return nil
+	}
+}
+
+func getLoadBalancerInterfaceInitialUpdateOpts(interfaceConfig map[string]interface{}) *load_balancer_interfaces.UpdateOpts {
+	updateInterfaceOpts := load_balancer_interfaces.UpdateOpts{}
+
+	if elem, ok := interfaceConfig["description"]; ok {
+		s := elem.(string)
+		updateInterfaceOpts.Description = &s
+	}
+
+	if elem, ok := interfaceConfig["ip_address"]; ok {
+		s := elem.(string)
+		updateInterfaceOpts.IPAddress = s
+	}
+
+	if elem, ok := interfaceConfig["name"]; ok {
+		s := elem.(string)
+		updateInterfaceOpts.Name = &s
+	}
+
+	if elem, ok := interfaceConfig["network_id"]; ok {
+		var networkID interface{}
+		networkID = elem
+		updateInterfaceOpts.NetworkID = &networkID
+	}
+
+	if elem, ok := interfaceConfig["virtual_ip_address"]; ok {
+		s := elem.(string)
+		if s != "" {
+			var virtualIPAddress interface{}
+			virtualIPAddress = s
+			updateInterfaceOpts.VirtualIPAddress = &virtualIPAddress
+		}
+	}
+
+	if elem, ok := interfaceConfig["virtual_ip_properties"].([]interface{}); ok && len(elem) == 1 {
+		updateInterfaceOpts.VirtualIPProperties = &load_balancer_interfaces.VirtualIPProperties{}
+		m := elem[0].(map[string]interface{})
+		updateInterfaceOpts.VirtualIPProperties.Protocol = m["protocol"].(string)
+		updateInterfaceOpts.VirtualIPProperties.Vrid = m["vrid"].(int)
+	}
+	return &updateInterfaceOpts
+}
+
+func getLoadBalancerInterfaceChanges(om map[string]interface{}, nm map[string]interface{}) *load_balancer_interfaces.UpdateOpts {
```

Comment:
prefixにexpandをつけて欲しいです。またom, nmが呼んでいるうちに何を指すのか混乱するのでもう少しわかりやすい変数名をつけてください。
```suggestion
func expandLoadBalancerInterfaceChanges(old map[string]interface{}, new map[string]interface{}) *load_balancer_interfaces.UpdateOpts {
```

Reply:
getではなくexpandに変更し、関数内の変数名を全体的に分かりやすくしました。